.github/workflows/code_coverage.yml

```diff
@@ -56,15 +56,23 @@ jobs:
         # See https://cmake.org/cmake/help/latest/manual/ctest.1.html for more detail
         run: ctest -C ${{env.BUILD_TYPE}}
 
-      - name: Gcov
-        # Run GCov target to collect coverage information.
-        run: cmake --build ${{github.workspace}}/build --config ${{env.BUILD_TYPE}} --target gcov
+      - name: Fastcov
+        # Run Fastcov target to collect coverage information.
+        run: cmake --build ${{github.workspace}}/build --config ${{env.BUILD_TYPE}} --target coverage
 
       - name: Lcov
         # Run LCov target to generate code coverage report
-        run: cmake --build ${{github.workspace}}/build --config ${{env.BUILD_TYPE}} --target lcov
+        run: cmake --build ${{github.workspace}}/build --config ${{env.BUILD_TYPE}} --target coverage-lcov
+
+      - name: Cobertura
+        # Run Cobertura target to generate code coverage report in XML
+        run: cmake --build ${{github.workspace}}/build --config ${{env.BUILD_TYPE}} --target coverage-cobertura
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v4.4.1
+        env:
+          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
@jbcoe This will need retrieving from the codecov website and see in the CI secrets

Reply:
I added `CODECOV_TOKEN` to repository secrets.

Reply:
Now this needs setting to resolve the final error:

```
error - 2025-01-01 19:56:18,903 -- Upload failed: {"message":"Token required because branch is protected"}
```